### DISCUSSION TITLE
Add LawnMowerActivity.RETURNING to lawn mower entity

### DISCUSSION THREAD

**Discussion by mikey0000** [home-assistant/architecture#1123] [Entity Models] [answered]
This can be used by husqvarna for example as it has

    MowerActivities.LEAVING,
    MowerActivities.GOING_HOME
    
 and be used by the HACS mammotion integration I'm also working on. I'm sure there is perhaps one more valuable one to have such as UPDATING (firmware) or READY (when not docked but not doing anything.) but mostly I'm really keen to get RETURNING in for a start.
 
 Thanks
 Adding PRs here:
 https://github.com/home-assistant/frontend/pull/21740
 https://github.com/home-assistant/core/pull/124261
https://github.com/home-assistant/developers.home-assistant/pull/2280
 
 nothing to be done for homeassistant.io repository from what I can see.

**Comment by cociweb** [comment]
Duplicate of https://github.com/home-assistant/architecture/discussions/1121

**Reply by mikey0000** [reply]
I disagree, what your asking for is much more extensive. I also agree with Martin that start can function as resume / pause.
I'm only asking for a few more activity states, not additional controls and the whole 9 yards.

I think my request would get lost in the discourse of your existing discussions.

**Reply by MartinHjelmare** [reply]
Yes, it's best to try and keep each discussion as scoped down as possible, to speed up the discussion resolution.

**Reply by cociweb** [reply]
> Yes, it's best to try and keep each discussion as scoped down as possible, to speed up the discussion resolution.

[off] Well, fully disagree, as it does not help.... just check, the climate's SwingH attribute. pretty simple -> 3 years ;)

**Comment by MartinHjelmare** [comment, marked as answer]
I think this addition is reasonable in general. It would be nice to scan the market a bit more though and try to find more devices (more than two) that support this in one way or another, so that we know that we're adding the feature in the right way.

**Reply by mikey0000** [reply]
Can you give me some guidance please? There are a lot of RTK based mowers now, trying to see what states they have is kinda difficult without either looking for existing libraries or reversing their apps. Don't get me wrong I'm fine with doing the research I just want to make sure I'm getting the correct thing your looking for.

**Reply by MartinHjelmare** [reply]
API docs, 3rd party libraries, application UI screenshots, data sheets, marketing material could be things that could show support.

**Reply by mikey0000** [reply]
Awesome, I'll get onto that!

**Reply by MartinHjelmare** [reply]
Ecovacs supports it too, by the way.

https://github.com/home-assistant/core/blob/94af95c95b1e5c24edcc754deee129ce0d778e94/homeassistant/components/ecovacs/lawn_mower.py#L30

**Reply by cociweb** [reply]
This is still valid here too (for tuya):
![image](https://github.com/user-attachments/assets/f0b2449c-5403-442d-9541-ef0475ab687d)

**Reply by MartinHjelmare** [reply]
Which of those activities means "returning to dock" (the feature addition we're discussing) for Tuya?

**Reply by cociweb** [reply]
"PARK" as it is in my [PR](https://github.com/home-assistant/core/pull/123093/files#diff-8a6f2eb8f2fa2534f53d3f0a193ba8b973654b84bed62d7b651f49645cffc4f3R33-R34)

**Reply by cociweb** [reply]
and of course you can clearly see the difference of the standby and paused.... ;) which are the result of stop vs pause.
and the 9th yard is just the fixed/spot mowing function. so still really don't understand... :)

**Reply by cociweb** [reply]
> UPDATING (firmware) or READY (when not docked but not doing anything.) 

from the 7th yards it is UPDATA and the 8th is STANDBY vs PAUSE

**Reply by cociweb** [reply]
In the linked ecovacs integration, the IDLE and PAUSED states are also differentiated (but unfortunatelly masked so information get lost to the user).

**Reply by mikey0000** [reply]
https://websiteoss.ecoflow.com/cms/upload/2023/4/26/EcoFlow%20Blade-User%20Manual%20V2.0%20(EN)_1682491239157.pdf

page 29 for the ecoflow blade (return to dock) plenty of info there as well for the other states.

**Reply by mikey0000** [reply]
https://navimow.segway.com/global/product/navimow-i.html
navimow advertisement material shows home button on their app picture.

**Reply by mikey0000** [reply]
Worx app has home button / return home
https://www.google.com/search?q=Worx+Landroid+Series+return+to+dock+button+app&tbm=isch&ved=2ahUKEwiPxqCpi-yHAxX65TgGHefiIEwQ2-cCegQIABAA&oq=Worx+Landroid+Series+return+to+dock+button+app&gs_lp=EgNpbWciLldvcnggTGFuZHJvaWQgU2VyaWVzIHJldHVybiB0byBkb2NrIGJ1dHRvbiBhcHBIpxJQqwNYkBFwAHgAkAEBmAHCAqAB_xCqAQcwLjkuMS4xuAEDyAEA-AEBigILZ3dzLXdpei1pbWeIBgE&sclient=img&ei=PDy4Zo-4NvrL4-EP58WD4QQ&bih=1390&biw=2452&client=ubuntu-sn&hs=lc4#imgrc=oxP0jF5H4uzqNM

https://github.com/mhaack/pylandroidcloud/blob/master/pylandroidcloud/states.py

**Reply by mikey0000** [reply]
@MartinHjelmare how many more do you need?

**Reply by MartinHjelmare** [reply]
I think we have enough for now.

**Comment by emontnemery** [comment]
The proposal is accepted, please go ahead with a PR implementing it @mikey0000 👍

**Reply by mikey0000** [reply]
Cool and I take it specifically returning, correct? Better to be as clear as possible 😁

**Reply by gjohansson-ST** [reply]
Yes. `LawnMowerActivity.RETURNING`

**Reply by mikey0000** [reply]
Added PRs to description :)

**Comment by gjohansson-ST** [comment]
Closing as has been implemented
